db-updater/main.py

```diff
@@ -480,6 +491,54 @@ def process_keepalive_message(data: dict) -> None:
     print(f'Based on keepalive["pitr"], we are {behind} behind realtime')
 
 
+def clear_dest_history(flight) -> None:
+    """Remove id from dest_history"""
+    global dest_history
+    if flight in dest_history:
+        dest_history.pop(flight)
+    return
+
+
+def check_for_diversions(data: dict) -> None:
+    """ETMS message, check destination"""
+    if "dest" in data and "id" in data:
+        global dest_history
+
+        dest = data.get("dest")
+        if dest == "":
+            return
+
+        flight = data.get("id")
+        if flight == "":
+            return
+
+        if flight in dest_history:
+            orig_dest = dest_history.get(flight)
+            if orig_dest != "":
+                if orig_dest != dest:
+                    data["diverted"] = True
+
+        dest_history[flight] = dest
+
+    return
+
+def check_for_new_destination(data: dict) -> None:
+    """Look for a new destination that hasn't been seen by an ETMS message yet"""
+    if "dest" in data and "id" in data:
+        global dest_history
+
+        flight = data.get("id")
+        if flight == "" or flight in dest_history:
+            return
+
+        dest = data.get("dest")
+        if dest == "":
+            return
+
+        dest_history[flight] = dest
+
+    return
```

Comment:
```suggestion
    global dest_history
    dest = data.get("dest")
    flight = data.get("id")

    if dest and flight and flight not in dest_history:
        dest_history[flight] = dest

    return
```
Similar simplification as above.